contracts/QuestNFT.sol

```diff
@@ -7,12 +7,46 @@ import "@openzeppelin/contracts/security/Pausable.sol";
 
 contract QuestNFT is ERC721, Ownable, Pausable {
     // Quest Registry
+    struct Quest {
+        uint256 questId;
+        address questCreator;
+        uint256 prerequisiteQuest;
+        uint256 questRewardXP;
+        address[] questPlayers;
```

Comment:
could be a mapping for more efficient lookups to check if someone is a player...idk if mappings can go in structs though